lib/eventlog/src/evtstr.c

```diff
@@ -89,105 +90,135 @@ evt_str_append_escape_bs(EVTSTR *es,
                          char escape_char)
 {
   /* a single character is escaped to at most 4 characters: \xXX */
+  size_t escaped_char_max_length = 4;
 
-  char *buf = (char *)alloca(4*unescaped_len + 1);
+  char escaped_buffer[128];
+  size_t escaped_buffer_capacity = sizeof(escaped_buffer);
+  size_t escaped_buffer_length = 0;
 
-  int i, dst;
+  int i;
 
-  for (i = 0, dst = 0; i < unescaped_len; i++)
+  for (i = 0; i < unescaped_len; i++)
     {
       unsigned c = (unsigned) unescaped[i];
 
       if (c < 32 && c != '\t')
         {
-          sprintf(&buf[dst], "\\x%02x", (unsigned char) unescaped[i]);
-          dst += 4;
+          sprintf(escaped_buffer + escaped_buffer_length, "\\x%02x", (unsigned char) unescaped[i]);
+          escaped_buffer_length += 4;
         }
       else if (unescaped[i] == escape_char)
         {
-          buf[dst++] = '\\';
-          buf[dst++] = escape_char;
+          escaped_buffer[escaped_buffer_length++] = '\\';
+          escaped_buffer[escaped_buffer_length++] = escape_char;
         }
       else
         {
-          buf[dst++] = unescaped[i];
+          escaped_buffer[escaped_buffer_length++] = unescaped[i];
+        }
+
+      size_t escaped_buffer_remaining_space = escaped_buffer_capacity - escaped_buffer_length;
```

Comment:
I would be cautious with `size_t - size_t` because `size_t` is `unsigned`.
It is possible that in the current iteration there is enough data, but in the next iteration it is possible that there isn't, and it is going to write possible uninitialized memory.
I would check if there is enough memory to write the next data (estimate the next data with max possible length: 4), and if not append to the buffer.

Reply:
If I understand it correctly, the condition below does the same thing you asked for.
The condition checks the availability for the next iteration, not for the current one.

```
if (escaped_buffer_remaining_space < escaped_char_max_length)
```

Reply:
My bad. You are right about it. It is correct.

Reply:
I'll rephrase the condition as we agreed IRL, it's more "natural" this way:

```
 if (escaped_buffer_capacity < escaped_buffer_length + escaped_char_max_length)
```

Reply:
Done.